src/ApiCallable.php

```diff
@@ -40,22 +39,43 @@
  */
 class ApiCallable
 {
-    const GRPC_CALLABLE_PARAM_COUNT = 3;
-    const GRPC_CALLABLE_METADATA_INDEX = 1;
-    const GRPC_CALLABLE_OPTION_INDEX = 2;
-    const GRPC_RESPONSE_STATUS_INDEX = 1;
+    // One Platform API codes
+    const STATUS_OK = 0;
+    const STATUS_CANCELLED = 1;
+    const STATUS_UNKNOWN = 2;
+    const STATUS_INVALID_ARGUMENT = 3;
+    const STATUS_DEADLINE_EXCEEDED = 4;
+    const STATUS_NOT_FOUND = 5;
+    const STATUS_ALREADY_EXISTS = 6;
+    const STATUS_PERMISSION_DENIED = 7;
+    const STATUS_RESOURCE_EXHAUSTED = 8;
+    const STATUS_FAILED_PRECONDITION = 9;
+    const STATUS_ABORTED = 10;
+    const STATUS_OUT_OF_RANGE = 11;
+    const STATUS_UNIMPLEMENTED = 12;
+    const STATUS_INTERNAL = 13;
+    const STATUS_UNAVAILABLE = 14;
+    const STATUS_DATA_LOSS = 15;
+    const STATUS_UNAUTHENTICATED = 16;
+
+    // Retry config
+    const CALLABLE_PARAM_COUNT = 3;
```

Comment:
These indexes are necessary only because we want to construct the data for the grpc stub. So these are a good candidate to be pulled out and handled in the grpc implementation of the transport interface.

Reply:
Agreed! This and also the creation of the streaming calls, so that exceptions can be thrown for `Json` stubs.